components/feature/media/src/main/java/mozilla/components/feature/media/service/MediaSessionServiceDelegate.kt

```diff
@@ -144,32 +145,18 @@ internal class MediaSessionServiceDelegate(
                 .build())
     }
 
-    private suspend fun updateNotification(sessionState: SessionState?) {
-        controller = sessionState?.mediaSessionState?.controller
-        val notificationId = SharedIdsHelper.getIdForTag(
-            context,
-            AbstractMediaSessionService.NOTIFICATION_TAG
-        )
-
-        val notification = notification.create(sessionState, mediaSession)
-
-        // Android wants us to always, always, ALWAYS call startForeground() if
-        // startForegroundService() was invoked. Even if we already did that for this service or
-        // if we are stopping foreground or stopping the whole service. No matter what. Always
-        // call startForeground().
+    private fun startForegroundNotification() {
+        val notification = notification.createDummy(mediaSession)
         service.startForeground(notificationId, notification)
+    }
 
-        val mediaSessionState = sessionState?.mediaSessionState
-        if (mediaSessionState != null &&
-            mediaSessionState.playbackState != MediaSession.PlaybackState.PLAYING) {
-            service.stopForeground(false)
+    private suspend fun updateNotification(sessionState: SessionState?) {
+        val notification = notification.create(sessionState, mediaSession)
 
-            NotificationManagerCompat.from(context)
-                .notify(notificationId, notification)
-        }
+        sessionState?.mediaSessionState?.let {
+            controller = it.controller
 
-        if (mediaSessionState == null) {
-            service.stopForeground(true)
+            NotificationManagerCompat.from(context).notify(notificationId, notification)
```

Comment:
I wonder why we remove the previous code that took care of stopping the foreground state if the media was not playing anymore?

Testing in sample browser something seems wrong: When playing media the notification is no longer sticky and can be swiped away - that's not how it should work and I wonder if we even are in foreground with that?

Just checked and somehow this behavior is even in Firefox Nightly... something seems wrong here?

Reply:
Looks like this is an Android 10 behaviour.  On Android 9 I can't swipe the notification away.  However, even when we swiped it away no crash happens.

Reply:
I've updated it so we now behave like Chrome.  Allowed to be swiped away when paused.